### DISCUSSION TITLE
Invertible conditional flows

### DISCUSSION THREAD

**Discussion by kirtanp** [facebookincubator/flowtorch#84] [Q&A] [answered]
### Issue Description (feature request)
Does flowtorch already support invertible conditional flows? That is, learning p(x | z). And given x and z, we get back the value sampled from the base distribution. Is there a reasonable way to do this already and if not, is it planned be supported any time soon? Thanks!

**Comment by stefanwebb** [comment, marked as answer]
@kirtanp thanks for the great question! First I can say that adding conditional flows is one of the highest items on our implementation agenda. We had to take it out after a major refactor and haven't gotten around to putting it back in yet. 

Separately, you can get the value sampled from the base distribution for analytically invertible flows with the `.normalize()` and `.rnormalize()` methods of `Flow`, which work independently from conditionality. See here: https://github.com/facebookincubator/flowtorch/blob/main/flowtorch/distributions/flow.py

**Reply by kirtanp** [reply]
Okay, thanks for the answer! I will use pyro for now then. Looking forward to using flowtorch soon. I do have a question about inversion in the pyro implementation, but I have asked it [here](https://forum.pyro.ai/t/inverting-a-conditional-transform-in-pyro/3739) on the forum to keep it separate.

**Comment by francesco-vaselli** [comment]
Hey, thanks for the great package! Really appreciate the effort 

As a followup on this thread I was wondering whether to this date conditional flows have been implemented or if there are plans for doing it in the near term.
Best regards! :)
